### ISSUE TITLE
Kernos.score silently returns NaN on constant-y targets; behaviour is undocumented

### ISSUE THREAD

**Issue by sachncs** [sachncs/kernos#15]
## Problem

`Kernos.score(X, y)` (and `kernos.bench.metric.r2`) returns `nan` when `y` is constant (i.e. `np.var(y) == 0`), but this edge case is undocumented. From `kernos/bench/metric.py:30-33`:

```python
ss_tot = float(np.sum((y - np.mean(y)) ** 2))
if ss_tot == 0.0:
    return float("nan")
```

`Kernos.score` calls `r2(np.asarray(y), self.predict(X))` and returns the float. A sklearn user expects `score` to be a finite number on any valid input; silently returning `nan` will propagate into GridSearchCV's `best_score_`, hyperparameter-search logs, and downstream metrics without warning.

The `nan` was added intentionally ("a trivially-zero model is not silently scored as perfect"), but the contract should be:
- Either return `0.0` for a constant target (the standard convention in some regression toolkits).
- Or raise `ValueError("y is constant; R^2 is undefined")`.
- At minimum, document the `nan` behaviour next to the score method and add a test.

## Evidence

- `kernos/bench/metric.py:24-34` — `r2` returns `nan` when `ss_tot == 0.0`.
- `kernos/estimator.py:178-182` — `Kernos.score` is documented only as `"""Return the R² score on (X, y)."""` with no mention of `nan`.
- `README.md` claims "R^2 score on held-out test set: 0.3…" with no caveat for constant-y cases.
- `docs/faq.md:108-115` discusses conditioning warnings but not constant-y.

## Root Cause

The behaviour was intentional but the API contract doesn't surface it.

## Impact

- Hyperparameter tuning against a constant target produces `best_score_ = nan` and silent search failures.
- Downstream metric pipelines (`rmse`, `mae` are still finite) look healthy while `r2` reports `nan` for the same prediction.

## Recommendation

- Document the behaviour explicitly in the `score` and `r2` docstrings, including a one-line example.
- Either return `0.0` (the "0% of variance explained" convention) or raise `ValueError` — pick one and apply consistently in both `Kernos.score` and `bench.metric.r2`.
- Add a regression test in `tests/unit/test_predict.py` that exercises both paths.

## Acceptance Criteria

- [ ] `r2(constant_y, predictions)` returns either `0.0` or raises `ValueError`, not `nan`.
- [ ] `Kernos.score(X, constant_y)` matches that behaviour.
- [ ] The `score` and `r2` docstrings describe the chosen behaviour.
- [ ] A unit test covers the constant-y case.

## Verification

```python
from kernos.bench.metric import r2
import numpy as np
y = np.zeros(10); yhat = np.zeros(10)
# After fix: assert r2(y, yhat) == 0.0 or pytest.raises(ValueError)
```

## Dependencies

None.

## Risk

If we change to raise, any downstream user who relied on the `nan` propagation will need to handle the new exception. Returning `0.0` is the lower-risk direction.

**Comment by sachncs** [comment]
Resolved in PR #27.
